### DISCUSSION TITLE
live editing

### DISCUSSION THREAD

**Discussion by Neon22** [kkinder/puepy#7] [General]
Can we use it inside the pyscript.com environment (coi set) to build components live ? like this(using the ltk editor):
https://discord.com/channels/972017612454232116/973504848530120714/1254819407398572034

**Comment by kkinder** [comment]
Yup, the demos from the tutorial are on PyScript.com already:

https://pyscript.com/@kkinder/puepy-tutorial/latest

You can clone those and start editing.

**Comment by Neon22** [comment]
Yes - excellent. but not quite what I was meaning. My linked example uses the py-editor. Well actually Chris has his own, but same idea.

The advantage? of running it in the pyeditor is that as you edit the code live in the editor - it is run. So you see changes as you type. No refresh step.
I thought you might be interested to play with the idea for the purpose of making it easy? to create components or to just build the user's puepy page directly live.

Its possible Chris's specific editor is not as good because its hard to see around errors at the moment. but this might be easier in the pyscript py-editor. Anyway - just an idea I though might interest you.
Cheers...

**Reply by kkinder** [reply]
Oh, I understand now.

I have done a little bit of experimentation with py-editor. From the docs though, it only seems to support a manual run cycle -- it draws a little "run" button on hover or you can use shortcut keys.

I might dig directly into CodeMirror and see what I can do though. Something like you describe is how I want the docs to present examples: as editors you can update live.
